csharp/src/elements/Geometry/Polygon.cs

```diff
@@ -140,31 +325,124 @@ public bool Disjoint(Polygon polygon)
+        /// <returns>
+        /// Returns true if the two Polygons intersect at least once.
+        /// </returns>
+        public bool Intersects(Polygon polygon)
+        {
+            if (polygon == null)
+            {
+                return false;
+            }
+            var clipper = new Clipper();
+            var solution = new List<List<IntPoint>>();
+            clipper.AddPath(this.ToClipperPath(), PolyType.ptSubject, true);
+            clipper.AddPath(polygon.ToClipperPath(), PolyType.ptClip, true);
+            clipper.Execute(ClipType.ctIntersection, solution);
+            return solution.Count != 0;
+        }
+
+        /// <summary>
+        /// Tests if any of the supplied Polygons share areas with this Polygon.
+        /// </summary>
+        /// <param name="polygons">The list of Polygons to compare.</param>
+        /// <returns>
+        /// Returns true if any of the supplied Polygons intersects with this Polygon.
+        /// </returns>
+        public bool Intersects(IList<Polygon> polygons)
+        {
+            if (polygons == null)
             {
-                if (Clipper.PointInPolygon(vertex, thisPath) == 1)
+                return false;
+            }
+            var clipper = new Clipper();
+            var solution = new List<List<IntPoint>>();
+            clipper.AddPath(this.ToClipperPath(), PolyType.ptSubject, true);
+            foreach (Polygon polygon in polygons)
+            {
+                clipper.AddPath(polygon.ToClipperPath(), PolyType.ptClip, true);
+            }
+            clipper.Execute(ClipType.ctIntersection, solution);
+            if (solution.Count != 0)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Tests if the supplied 2D Vector3 point is on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="point">The 2D Vector3 point to compare.</param>
+        /// <returns>
+        /// Returns true if the supplied 2D Vector3 point coincides with the perimeter of this Polygon.
+        /// </returns>
+        public bool Touches(Vector3 point)
+        {
+            var scale = 1024.0;
+            var thisPath = this.ToClipperPath();
+            var intPoint = new IntPoint(point.X * scale, point.Y * scale);
+            if (Clipper.PointInPolygon(intPoint, thisPath) != -1)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if all the 2D Vector3 points in the supplied collection are on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="points">The 2D Vector3 point to compare.</param>
+        /// <returns>
+        /// Returns true if all the supplied 2D Vector3 points coincide with the perimeter of this Polygon.
+        /// </returns>
+        public bool Touches(IList<Vector3> points)
```

Comment:
As above, use singular `Touches` in this method.

Reply:
Right-o

Reply:
Done.